engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/AnalyzeModuleInScopeJob.scala

```diff
@@ -60,8 +61,13 @@ final class AnalyzeModuleInScopeJob(
         exports = ModuleExportsDiff.compute(prevExports, newExports),
         updates = SuggestionDiff.compute(Tree.empty, newSuggestions)
       )
-      sendModuleUpdate(notification)
-      module.setIndexed(true)
+      if (!ctx.state.suggestions.markAsIndexed(module)) {
```

Comment:
The problem I see [continues to be the same](https://github.com/enso-org/enso/pull/9639#discussion_r1556904725). Line 44 deals with `state.suggestions` and line 64 does that as well. But meanwhile anything can happen. What if a new request to `markAsNotIndexed` comes in another thread?

> there was a reason why logic was added to Analyze*Job to deal with this scenario

My experience with Enso code base and its _readiness for concurrent execution_ doesn't support claims that advocate _reason_ and/or _logic_. Even if they were present, it is questionable they were used in a _sound way_. Except using the most advanced concurrent building blocks, I only see attempts to avoid answers to the basic question: _how that is supposed to work at all_?